src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -100,6 +114,109 @@ pub struct RSAggregateResult {
     pub type_mask: RSResultTypeMask,
 }
 
+impl RSAggregateResult {
+    /// Create a new aggregate result with th given capacity
+    pub fn new(cap: usize) -> Self {
+        Self {
+            num_children: 0,
+            children_cap: cap as c_int,
+            children: Self::new_children(cap),
+            type_mask: RSResultTypeMask::empty(),
+        }
+    }
+
+    fn push(&mut self, result: &mut RSIndexResult) {
+        if self.num_children >= self.children_cap {
+            self.grow()
+        }
+
+        // SAFETY: `num_children` is within the valid range of the `children` pointer, else we would
+        // have grown above to make it valid.
+        let slot = unsafe { self.children.add(self.num_children as _) };
+
+        // SAFETY: we just used `add` to ensure `slot` is a valid pointer and is "aligned correctly"
```

Comment:
This actually assumes that `result` stays allocated during the whole `RSAggregateResult` lifetime, right?

Reply:
Not great for safety but that will go away as we keep porting code to Rust.

Maybe make it clearer in the function doc as I'm sure we'll miss that at some point when using this. :)